packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -466,4 +503,18 @@ export class TransformerResolver implements TransformerResolverProvider {
       });
     }
   }
+
+  private stashStatementGenerator(
+    runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty,
+  ): (name: string, value?: string, object?: string) => string {
+    const jsStashStatement = (name: string, value?: string, object?: string): string => {
```

Comment:
1. Are we ever going to need the type signature of a `stash` statement elsewhere, suggesting that we should create a typealias and/or structured input for it?
2. naming nit: these are generator functions, not statements :)

Reply:
Great feedback, thanks. Extracting to types was borderline necessary already but definitely necessary after introducing `stashString` and `stashExpression`. Addressed in [113d921](https://github.com/aws-amplify/amplify-category-api/pull/2812/commits/113d921c23ec8d9262af4a05dca7705b881dcb82)